### PR TITLE
Respecting filter in WLog_GetLogLevel

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by bmiklautz** [comment]
Did some local tests. The filters work now as expected. However we now have another function call for each wlog messages. Anyway +1. Thanks for fixing this.

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 1]
```diff
@@ -295,14 +296,14 @@ BOOL WLog_PrintMessage(wLog* log, wLogMessage* message, ...)
...
-	}
-	else
-	{
-		return log->Level;
-	}
+	if (log->FilterLevel < 0)
+		log->FilterLevel = WLog_GetFilterLogLevel(log);
+
+	if ((log->FilterLevel >= 0) && (log->FilterLevel != WLOG_LEVEL_INHERIT))
+		return log->FilterLevel;
```
Comment (mfleisz) [inline winpr/libwinpr/utils/wlog/wlog.c]: missing tab

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 2]
```diff
@@ -564,11 +565,11 @@ int WLog_GetFilterLogLevel(wLog* log)
 	}
 
 	if (match)
-	{
-		iLevel = (int) g_Filters[i].Level;
-	}
+		log->FilterLevel = g_Filters[i].Level;
```
Comment (mfleisz) [inline winpr/libwinpr/utils/wlog/wlog.c]: missing tab

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 3]
```diff
@@ -564,11 +565,11 @@ int WLog_GetFilterLogLevel(wLog* log)
...
+	else
+		log->FilterLevel = WLOG_LEVEL_INHERIT;
```
Comment (mfleisz) [inline winpr/libwinpr/utils/wlog/wlog.c]: missing tab

**Comment by mfleisz** [comment]
Aside from the formatting issues +1

File: winpr/libwinpr/utils/wlog/wlog.c (hunk 1 above)
Comment (akallabeth) [inline winpr/libwinpr/utils/wlog/wlog.c]: @mfleisz where?

**Comment by mfleisz** [comment]
@akallabeth Hm looks like github was fooling me around - looks everything ok now - sorry!

File: winpr/libwinpr/utils/wlog/wlog.c (hunk 3 above)
Comment (akallabeth) [inline winpr/libwinpr/utils/wlog/wlog.c]: @mfleisz again, where?

File: winpr/libwinpr/utils/wlog/wlog.c (hunk 2 above)
Comment (akallabeth) [inline winpr/libwinpr/utils/wlog/wlog.c]: @mfleisz again, where?

**Comment by akallabeth** [comment]
@mfleisz Ah, ok :) was a bit surprised to see your comments ;)

**Pull request merged**